File: src/main/java/com/nospawnn/adventOfCode/lib/SolverMap.java
package com.nospawnn.adventOfCode.lib;

import java.util.HashMap;
import java.util.Map;

public class SolverMap {
    private static final Map<String, Solver> SOLVERS = new HashMap<>();

    static {
        SOLVERS.put("2023/01", new com.nospawnn.adventOfCode.y2023.Day01());
    }

    public static Solver getSolver(String name) {
        return SOLVERS.get(name);
    }
}

File: src/main/kotlin/runner/SolverMap.kt
package runner

import adventOfCode.lib.Solver
import runner.GetInput.getInputPath
import java.io.File
import java.nio.file.Files

object SolverMap {
    private val solvers: Map<String, Solver> = hashMapOf(
        // 2015
        "2015/01" to adventOfCode.y2015.d01.Solution(),
        "2015/02" to adventOfCode.y2015.d02.Solution(),

        // 2023
        "2023/01" to adventOfCode.y2023.d01.Solution()
    )

    fun getSolver(name: String): Solver? {
        return solvers[name]
    }

    fun solveAll() {
        solvers.toList().sortedBy { it.first }.forEach { (name, solver) ->
            val (year, day) = name.split("/");
            val inputFilePath = getInputPath(year, day)

            if (!Files.exists(inputFilePath)) return@forEach

            val f = File(inputFilePath.toUri())
            val input = f.readText()

            solver.solve(input);
        }
    }
}
